src/main/java/org/spongepowered/api/plugin/PluginManager.java

```diff
@@ -77,4 +79,17 @@
      */
     boolean isLoaded(String id);
 
+    /**
+     * Lets the given plugin modify the world generator. Does nothing if the
+     * plugin does not implement the {@link WorldGeneratorProvider} interface.
+     *
+     * @param plugin The plugin to get the world generator from.
+     * @param worldName Name of the world.
+     * @param settings A string with (usually user-supplied) settings for the
+     *        world generator plugin, it's up to the world generator plugin to
+     *        do anything with this.
+     * @param worldGenerator The world generator to modify.
+     */
+    void modifyWorldGenerator(PluginContainer plugin, String worldName, String settings, WorldGenerator worldGenerator);
```

Comment:
The settings should probably be changed to be a DataContainer as well, it would be easier for plugins to use as it doesn't require knowledge of the setting's format.